### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by mcmillanator**
Fixed the pin loop

### CODE REVIEW

File: assignment.rb [hunk 1]
```diff
@@ -27,24 +27,51 @@ def initialize()
...
+				puts options
+				print ">> "
+				@selected_option = gets.chomp.to_i
+			end
+			return true
+		end
+		return @return
+	end
+
+	def login()
```
Comment (rpearce) [inline assignment.rb]: Just an FYI: if you have a method that doesn't accept any arguments, you don't need the `()`

**Comment by rpearce** [comment]
Just some general commentary:
This method
```ruby
def authenticate(user, pin)
  if @accounts.has_key? (user)
    if @accounts[user][0] == pin
      return true
    end
  end
end
```
 could be refactored first to
```ruby
def authenticate(user, pin)
  true if @accounts.has_key?(user) && @accounts[user][0] == pin
end
```
but since logical operators like `&&` returns `true`/`false`, you could just do
```ruby
def authenticate(user, pin)
  @accounts.has_key?(user) && @accounts[user][0] == pin
end
```

File: assignment.rb [hunk 2]
```diff
@@ -27,24 +27,51 @@ def initialize()
...
+					"Error, no pin entered\n\n"
+				).to_i
+
+					# opens my little surprise gif if the user and pin are not authenticated
 			if !@user.authenticate(@name, @pin)
 				puts `open http://replygif.net/i/981.gif 2> /dev/null`
 				puts `google-chrome http://replygif.net/i/981.gif 2> /dev/null`
 			end
 		end
+		@user_available_funds = @user.balance
```
Comment (rpearce) [inline assignment.rb]: I get this error when I enter `Stan` and `1234`
```ruby
Please Enter Your Name
>> Stan
Please Enter Your Pin
>> 1234
test.rb:74:in `login': undefined method `balance' for #<User:0x007fb622843de0> (NoMethodError)
	from test.rb:119:in `<main>'
```

File: assignment.rb [hunk 3]
```diff
@@ -27,24 +27,51 @@ def initialize()
...
+			@selected_option = nil
+			until (1..options.length).include?(@selected_option)
+				puts output
+				puts options
+				print ">> "
+				@selected_option = gets.chomp.to_i
+			end
+			return true
+		end
+		return @return
```
Comment (rpearce) [inline assignment.rb]: When it's the last line in a method, there's no need for a `return` statement

File: assignment.rb [hunk 4]
```diff
@@ -27,24 +27,51 @@ def initialize()
...
 	end
 
-	def login()
-		until @user.authenticate(@name, @pin)
-			until !@name.empty?
-				puts "Please Enter Your Name"
+	def prompt(output, err, options = nil)
+		@return = ""
+		if options.nil?
+			until !@return.empty?
```
Comment (rpearce) [inline assignment.rb]: ABP: Always be positive (if you can). Would this be easier to read than this?
```ruby
while @return.empty?
```

**Comment by mcmillanator** [comment]
Thank you for the suggestions.  I've used them all here and in 1.4.  I'm finding that the biggest thing you are teaching me is how to think.  Thank you very much for that!

**Comment by mcmillanator** [comment]
I fixed all of these issues in the master branch.  (oops)  That was weeks ago.  I have a much better workflow habits now.  I'm closing this merge and dropping this bugged branch.

**Pull request closed without merging**